src/main/java/com/laserfiche/repository/api/clients/impl/EntriesClientImpl.java

```diff
@@ -0,0 +1,777 @@
+                    }
+                    if (httpResponse.getStatus() == 401) {
+                        throw new RuntimeException("Access token is invalid or expired.");
+                    }
+                    if (httpResponse.getStatus() == 403) {
+                        throw new RuntimeException("Access denied for the operation.");
+                    }
+                    if (httpResponse.getStatus() == 404) {
+                        throw new RuntimeException("Request entry id not found.");
+                    }
+                    if (httpResponse.getStatus() == 429) {
+                        throw new RuntimeException("Rate limit is reached.");
+                    }
+                    return httpResponse.getBody();
+                });
+    }
+
+    @Override
+    public CompletableFuture<ODataValueOfIListOfFieldValue> assignFieldValues(String repoId, Integer entryId,
+            Map<String, FieldToUpdate> requestBody, String culture) {
+        Map<String, Object> queryParameters = new HashMap<>();
+        if (culture != null) {
+            queryParameters.put("culture", culture);
+        }
+        return httpClient
+                .put(baseUrl + "/v1/Repositories/{repoId}/Entries/{entryId}/fields")
+                .routeParam("repoId", repoId)
+                .routeParam("entryId", String.valueOf(entryId))
+                .queryString(queryParameters)
+                .contentType("application/json")
+                .body(requestBody)
+                .asObjectAsync(ODataValueOfIListOfFieldValue.class)
+                .thenApply(httpResponse -> {
+                    if (httpResponse.getStatus() == 400) {
+                        throw new RuntimeException("Invalid or bad request.");
+                    }
+                    if (httpResponse.getStatus() == 401) {
+                        throw new RuntimeException("Access token is invalid or expired.");
+                    }
+                    if (httpResponse.getStatus() == 403) {
+                        throw new RuntimeException("Access denied for the operation.");
+                    }
+                    if (httpResponse.getStatus() == 404) {
+                        throw new RuntimeException("Requested entry id not found.");
+                    }
+                    if (httpResponse.getStatus() == 423) {
+                        throw new RuntimeException("Entry is locked.");
+                    }
+                    if (httpResponse.getStatus() == 429) {
+                        throw new RuntimeException("Rate limit is reached.");
+                    }
+                    return httpResponse.getBody();
+                });
+    }
+
+    @Override
+    public CompletableFuture<CreateEntryResult> importDocument(PostEntryWithEdocMetadataRequest requestBody,
+            String repoId, Integer parentEntryId, String fileName, Boolean autoRename, String culture) {
```

Comment:
is the electronicDocument file parameter missing?

Reply:
Yes, it's not working. It is planned work in another work item (TFS).